### ISSUE TITLE
[SC2046] false positive from `return $([ "$(tput colors)" -ge "8" ])`

### ISSUE THREAD

**Issue by SwuduSusuwu** [koalaman/shellcheck#3119]
#### For bugs
- Rule Id: SC2046):
- My shellcheck version: https://github.com/SwuduSusuwu/SubStack/blob/experimental/.github/workflows/codacy.yml
- [ X ] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2046)
- [ X ] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/sh
SUSUWU_HAS_STANDARD_CONSOLE() ( #/* Usage: `if SUSUWU_HAS_STANDARD_CONSOLE; then $(SUSUWU_SH_USE "${SUSUWU_SH_<color>}" "<message>")` */                                                                                                                      if command -v "tput" >/dev/null; then #if installed `ncurses-utils`                                                             return $([ "$(tput colors)" -ge "8" ]) #standard console                                                                    fi                                                                                                                            for CONSOLE in "xterm" "xterm-color" "xterm-256color" "screen" "screen-color" "screen-256color" "tmux" "tmux-color" "tmux-256color" "linux" "rxvt" "rxvt-unicode"; do                                                                                         if [ "${TERM}" = "${CONSOLE}" ]; then                                                                                           return 0 #standard console
    fi                                                                                                                          done                                                                                                                          return 1                                                                                                                    )

```

#### Here's what shellcheck currently says:
> [Quote this to prevent word splitting.](https://github.com/SwuduSusuwu/SubStack/security/code-scanning/2251)

#### Here's what I wanted or expected to see:
How to add more quotes, or "Passed".

`return $("[ "$(tput colors)" -ge "8" ]")` is a syntax error, and `return "$([ "$(tput colors)" -ge "8" ])"` is a syntax error (both give command "not found"), so do not know how to add more quotes.

**Comment by brother** [comment]
I am not sure that code does what you want it to do but the correct quoting would be the complete subshell (second line in your report).

But let's break it down to try to understand what we have.
`return $([ "$(tput colors)" -ge "8" ])`

The `return` can be dropped first, we want to control the flow in the function and end on a exit code.

Starting inside the square brackets we have a subshell excuting `tput colors`. It will give us a exit code to work with.
That will be compared as being either greater than or equal to 8. That comparison will leave us with either 0 or 1.
```
brother ~$ [ "256" -ge "8" ]
brother ~$ echo $?
0
brother ~$ [ "8" -ge "8" ]
brother ~$ echo $?
0
brother ~$ [ "1" -ge "8" ]
brother ~$ echo $?
1
```

The square bracket comparison will the be passed up a step to the subshell and executed. In most cases I would assume that there are no command 0 nor 1 and thus you'll have "command not found". 

I think shellcheck does suggest the correct quoting in this case. I am doubting that the code presented is constructed correct.

Maybe
```
        if [ "$(tput colors)" -ge "8" ]; then
            return 0 #standard console
        fi
        return 1
```

**Comment by SwuduSusuwu** [comment]
>    if [ "$(tput colors)" -ge "8" ]; then
      return 0 #standard console
    fi
     return 1
 
is a suitable workaround. Too bad it is 4 lines instead of 1 to do this, such as `return [ "$(tput colors)" -ge "8" ]`

**Comment by SwuduSusuwu** [comment]
Workaround:
```
test "$(tput colors)" -ge "8"                                                                                                 return $?
```
reduces those 4 lines to 2.
